module/signature/signer_indices.go

```diff
@@ -0,0 +1,319 @@
+package signature
```

Comment:
I think the tools fo signer indices dealing with staking and beacon signatures are very specific to Hotstuff and could live under `consensus/hotstuff/` instead of `module/signature`.

Reply:
You are right. That's what I was doing initially. 

Later we realized the signer indices is not used just by consensus, but also to encode the approver IDs in the seal, which is not hotstuff related. 

If we keep non-beacon related functions in this package to be reused, and move beacon-related functions to `consensus/hotstuff`, then we would have 2 different places about Encoding/Decoding signer indices, feels a bit having similar logic all over the places. 

So in the end, decided to just keep all signer indices related encoding/decoding functions here. It's a trade-off.